keras_cv/models/densenet.py

```diff
@@ -0,0 +1,306 @@
+            the 4D tensor output of the
+            last convolutional block.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional block, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+      classes: optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
+      classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+
+    Returns:
+      A `keras.Model` instance.
+    """
+    if not (weights in {None} or tf.io.gfile.exists(weights)):
+        raise ValueError(
+            "The `weights` argument should be either "
+            "`None` or the path to the weights file to be loaded."
+        )
+
+    if include_top and None in input_shape:
+        raise ValueError(
+            "If `include_top` is True, "
+            "you should specify a static `input_shape`. "
+            f"Received: input_shape={input_shape}"
+        )
+
+    # Determine proper input shape
+    img_input = utils.get_input_tensor(input_shape, input_tensor)
+
+    if include_preprocessing:
+        x = layers.Rescaling(1 / 255.0)
+
+    x = layers.ZeroPadding2D(padding=((3, 3), (3, 3)))(img_input)
+    x = layers.Conv2D(64, 7, strides=2, use_bias=False, name="conv1/conv")(x)
+    x = layers.BatchNormalization(axis=BN_AXIS, epsilon=1.001e-5, name="conv1/bn")(x)
+    x = layers.Activation("relu", name="conv1/relu")(x)
+    x = layers.ZeroPadding2D(padding=((1, 1), (1, 1)))(x)
+    x = layers.MaxPooling2D(3, strides=2, name="pool1")(x)
+
+    x = dense_block(x, blocks[0], name="conv2")
+    x = transition_block(x, 0.5, name="pool2")
+    x = dense_block(x, blocks[1], name="conv3")
+    x = transition_block(x, 0.5, name="pool3")
+    x = dense_block(x, blocks[2], name="conv4")
+    x = transition_block(x, 0.5, name="pool4")
+    x = dense_block(x, blocks[3], name="conv5")
+
+    x = layers.BatchNormalization(axis=BN_AXIS, epsilon=1.001e-5, name="bn")(x)
+    x = layers.Activation("relu", name="relu")(x)
+
+    if include_top:
+        x = _apply_classifier(x, classes, classifier_activation)
+    else:
+        x = _apply_pooling_layer(x, pooling)
+
+    if input_tensor is not None:
+        inputs = keras.utils.get_source_inputs(input_tensor)
+    else:
+        inputs = img_input
+
+    model = keras.Model(inputs, x, name=_get_name(blocks))
+
+    if weights is not None:
+        model.load_weights(weights)
+    return model
+
+
+def DenseNet121(
+    include_preprocessing,
+    include_top=True,
+    weights="imagenet",
```

Comment:
It seems that the imagenet will fail since the base class expect a file path.

Reply:
Yep - made a mistake leaving this as default.

Reply:
Thanks!